lms/djangoapps/instructor_task/api_helper.py

```diff
@@ -190,6 +240,27 @@ def _update_instructor_task(instructor_task, task_result):
             instructor_task.save()
 
 
+def _update_instructor_task_state(instructor_task, task_state, message=None):
+    """
+     Update state and output of InstructorTask object.
+    """
+    instructor_task.task_state = task_state
+    if message:
+        instructor_task.task_output = message
+
+    instructor_task.save()
+
+
+def _handle_instructor_task_failure(instructor_task, error):
+    """
+    Do required operations if task creation was not complete.
+    """
+    _update_instructor_task_state(instructor_task, FAILURE, error.message)
+    log.info("instructor task (%s) failed, result: %s", instructor_task.task_id, error.message)
```

Comment:
Can we log before updating the task state?  Reason is because logging is less likely to fail than SQL model updates.  And since we're already in the exception handling path here, it's more possible that the database is down than the log-tracker.